dao-macros/src/serialization.rs

```diff
@@ -110,3 +107,103 @@ pub fn derive_to_bytes(input: TokenStream) -> TokenStream {
 
     TokenStream::from(expanded)
 }
+
+fn variants(input: DeriveInput) -> Result<Vec<Ident2>, TokenStream> {
+    let fields = match &input.data {
+        Data::Enum(DataEnum { variants, .. }) => {
+            variants.into_iter().map(|x| &x.fields).collect::<Vec<_>>()
+        }
+        _ => {
+            return Err(TokenStream::from(
+                quote! { compile_error!("Expected an enum."); },
+            ))
+        }
+    };
+
+    for f in fields.into_iter() {
+        if *f != Fields::Unit {
+            return Err(TokenStream::from(
+                quote! { compile_error!("Expected an enum with unit variants."); },
+            ));
+        }
+    }
+
+    let variants = match input.data {
+        Data::Enum(DataEnum { variants, .. }) => {
+            variants.into_iter().map(|x| x.ident).collect::<Vec<_>>()
+        }
+        _ => {
+            return Err(TokenStream::from(
+                quote! { compile_error!("Expected an enum."); },
+            ))
+        }
+    };
+    Ok(variants)
+}
+
+pub fn derive_to_bytes_enum(input: DeriveInput) -> TokenStream {
+    let enum_ident = input.ident.clone();
+    let variants = match variants(input) {
+        Ok(variants) => variants,
+        Err(error_stream) => return error_stream,
+    };
+
+    let mut append_bytes = TokenStream2::new();
+    append_bytes.append_all(variants.iter().enumerate().map(|(index, ident)| {
+        let index: u32 = index as u32 + 1;
+        quote! {
+          #enum_ident::#ident => #index,
+        }
+    }));
+
+    let expanded = quote! {
+      impl casper_types::bytesrepr::ToBytes for #enum_ident {
+        fn serialized_length(&self) -> usize {
+          1
+        }
+
+        fn to_bytes(&self) -> Result<Vec<u8>, casper_types::bytesrepr::Error> {
+          let mut vec = Vec::with_capacity(self.serialized_length());
+          vec.append(
+            &mut match self {
+                #append_bytes
+              }
+              .to_bytes()?,
+          );
+          Ok(vec)
+        }
+      }
+    };
+
+    TokenStream::from(expanded)
+}
+
+pub fn derive_from_bytes_enum(input: DeriveInput) -> TokenStream {
+    let enum_ident = input.ident.clone();
+    let variants = match variants(input) {
+        Ok(variants) => variants,
+        Err(error_stream) => return error_stream,
+    };
+
+    let mut append_bytes = TokenStream2::new();
+    append_bytes.append_all(variants.iter().enumerate().map(|(index, ident)| {
+        let index: u32 = index as u32 + 1;
+        quote! {
+          #index => Ok((#enum_ident::#ident, bytes)),
+        }
+    }));
+
+    let expanded = quote! {
+      impl casper_types::bytesrepr::FromBytes for #enum_ident {
+        fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), casper_types::bytesrepr::Error> {
```

Comment:
I'm pretty sure that fully-qualified paths in the body of a macro should be applied to `Result` , `Option`, and the other parts of std either. It makes us sure that we eliminate possible name ambiguities.